packages/insomnia-components/src/dropdown/dropdown-item.tsx

```diff
@@ -98,10 +114,10 @@ export class DropdownItem extends PureComponent<DropdownItemProps> {
       return;
     }
 
-    if (this.props.hasOwnProperty('value')) {
-      onClick(this.props.value, event);
-    } else {
-      onClick(event);
+    if (isControlledInput<T>(this.props)) {
+      this.props.onClick?.(this.props.value, event);
+    } else if (isUncontrolledInput(this.props)) {
+      this.props.onClick?.(event);
```

Comment:
I tried to change this to the following, interesting how it causes type errors.

(This is not a suggestion, just showing the diff)
```suggestion
    if (isControlledInput<T>(this.props)) {
      onClick(this.props.value, event);
    } else if (isUncontrolledInput(this.props)) {
      onClick(event);
```

Reply:
I, as well, had to wrestle with it... A LOT.

This particular part, though, is because of a problem with TypeScript below 4.4.  In 4.4 they added https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#cfa-aliased-conditions which _I would expect_ solves this problem.